omnichain/staking/contracts/Staking.sol

```diff
@@ -29,74 +31,132 @@ contract Staking is ERC20, zContract {
         chainID = chainID_;
     }
 
+    modifier onlySystem() {
+        require(
+            msg.sender == address(systemContract),
+            "Only system contract can call this function"
+        );
+        _;
+    }
+
+    function bytesToBech32Bytes(
```

Comment:
It looks like this one could be `internal`

Reply:
967401c